kafka-client/src/main/java/dev/responsive/kafka/api/stores/TtlProvider.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright 2024 Responsive Computing, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package dev.responsive.kafka.api.stores;
+
+import dev.responsive.kafka.internal.utils.StateDeserializer;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import org.apache.kafka.common.serialization.Serde;
+
+public class TtlProvider<K, V> {
+
+  /**
+   * Creates a new TtlProvider with the given default duration to retain records for unless
+   * overridden. To allow ttl overrides for individual records, you can use one of the
+   * {@link #fromKey(Function, Serde)}, {@link #fromValue(Function, Serde)},
+   * or {@link #fromKeyAndValue(BiFunction, Serde, Serde)} methods to define the row-level
+   * override function.
+   *
+   * @return a new TtlProvider that will retain records for the specified default duration
+   */
+  public static <K, V> TtlProvider<K, V> withDefault(final Duration defaultTtl) {
+    return new TtlProvider<>(
+        TtlType.DEFAULT_ONLY,
+        TtlDuration.of(defaultTtl),
+        (ignoredK, ignoredV) -> Optional.empty(),
+        null,
+        null
+    );
+  }
+
+  /**
+   * Creates a new TtlProvider that has no default (equivalent to infinite retention for
+   * all records unless an override is specified).  Must be used in combination with
+   * exactly one of the {@link #fromKey(Function, Serde)}, {@link #fromValue(Function, Serde)},
+   * and {@link #fromKeyAndValue(BiFunction, Serde, Serde)} methods to define the row-level
+   * override function.
+   * <p>
+   * If no ttl whatsoever is desired for this store, simply don't configure a TtlProvider
+   * to begin with.
```

Comment:
same comment as above, just leaving this here as a reminder to change the docs if you agree